integration/benchmark/contLoadGenerator.go

```diff
@@ -239,50 +254,55 @@ func (lg *ContLoadGenerator) setupFavContract() error {
 	return nil
 }
 
-func (lg *ContLoadGenerator) Start() {
-	// spawn workers
-	for i := 0; i < lg.loadParams.TPS; i++ {
-		var worker Worker
-
-		// TODO(rbtz): create an interface for different load types: Start()
-		switch lg.loadParams.LoadType {
-		case TokenTransferLoadType:
-			worker = NewWorker(i, 1*time.Second, lg.sendTokenTransferTx)
-		case TokenAddKeysLoadType:
-			worker = NewWorker(i, 1*time.Second, lg.sendAddKeyTx)
-		case ConstExecCostLoadType:
-			worker = NewWorker(i, 1*time.Second, lg.sendConstExecCostTx)
-		// other types
-		default:
-			worker = NewWorker(i, 1*time.Second, lg.sendFavContractTx)
-		}
-
+func (lg *ContLoadGenerator) startWorkers(numWorkers, diff int) {
+	for i := 0; i < diff; i++ {
+		worker := NewWorker(numWorkers+i, 1*time.Second, lg.workFunc)
 		worker.Start()
-		lg.workerStatsTracker.AddWorker()
-
-		lg.workers = append(lg.workers, &worker)
+		lg.workers = append(lg.workers, worker)
 	}
-
-	lg.workerStatsTracker.StartPrinting(1 * time.Second)
 }
 
-func (lg *ContLoadGenerator) Stop() {
-	defer lg.log.Debug().Msg("stopped generator")
-
-	lg.stopped = true
+func (lg *ContLoadGenerator) stopWorkers(numWorkers, diff int) {
 	wg := sync.WaitGroup{}
-	wg.Add(len(lg.workers))
-	for _, w := range lg.workers {
-		w := w
+	wg.Add(diff)
 
-		go func() {
+	start := numWorkers - diff
+	for i := start; i < numWorkers; i++ {
+		go func(w *Worker) {
 			defer wg.Done()
-
 			lg.log.Debug().Int("workerID", w.workerID).Msg("stopping worker")
 			w.Stop()
-		}()
+		}(lg.workers[i])
 	}
 	wg.Wait()
+
+	lg.workers = lg.workers[:start]
+}
+
+func (lg *ContLoadGenerator) SetTPS(num uint) error {
```

Comment:
```suggestion
// SetTPS compares the given tps to the existing tps to determine whether to increase
// or decrease the load.
// it increases/decreases the load by adjusting the number of workers, since each worker
// is responsible for sending the load at 1 tps.
func (lg *ContLoadGenerator) SetTPS(num uint) error {
```